src/gui/KM_GUIMenuOptions.pas

```diff
@@ -402,13 +482,94 @@ procedure TKMMenuOptions.Show;
 end;
 
 
+
+procedure TKMMenuOptions.KeybindClick(Sender: TObject);
+begin
+  if ColumnBox_Options_Keys.TopIndex < 0 then Exit;
+
+  if Sender = Button_Options_Keys then
+    PopUp_Options_Keys.Show;
+
+  if Sender = Button_Options_Keys_OK then
+    PopUp_Options_Keys.Hide;
+
+  if Sender = Button_Options_Special_Keys then
+    PopUp_Options_Special_Keys.Show;
+
+  if Sender = Button_Options_Special_Keys_OK then
+    PopUp_Options_Special_Keys.Hide;
+end;
+
+
+procedure TKMMenuOptions.Keybind_ListKeySave(Key: Word; Shift: TShiftState);
+var
+  aID, I, D: Integer;
+begin
+  aID := ColumnBox_Options_Keys.ItemIndex;
+  D := 0;
```

Comment:
Could change it to 
if (Key = gResKeys.Keys[I]) or (Key in [121, 122]) then exit; 
and leave D out, I think.